model/project.rb

```diff
@@ -24,6 +25,12 @@ def user_ids
     access_tags_dataset.where(hyper_tag_table: Account.table_name.to_s).select_map(:hyper_tag_id)
   end
 
+  def has_valid_payment_method?
+    return true unless Config.managed_service
+    # TODO: More complex checks expiration date etc.
+    !!billing_info&.payment_methods&.any?
```

Comment:
not sure if payment methods should be part of billing info or projects directly.